File: shop_dao.py
import mysql.connector
import json

class ShopDAO:
    def __init__(self, config_path='config.json'):
        self.config = self.read_config(config_path)
    
    def read_config(self, config_path):
        with open(config_path, 'r') as file:
            config = json.load(file)
        return config['Database']

    def get_cursor(self):
        connection = mysql.connector.connect(
            host=self.config['host'],
            user=self.config['user'],
            password=self.config['password'],
            database=self.config['database']
        )
        mycursor = connection.cursor()
        return mycursor, connection
    
    def close_all(self, mycursor, connection):
        connection.close()
        mycursor.close()

    def create_order(self, product_id, quantity, total_price):
        mycursor, connection = self.get_cursor()

        # Insert a new order into the 'orderdata' table
        order_sql = '''
            INSERT INTO orderdata (ProductID, Quantity, TotalPrice) 
            VALUES (%s, %s, %s)
        '''
        values = (product_id, quantity, total_price)

        mycursor.execute(order_sql, values)
        connection.commit()

        # Close the cursor and connection
        self.close_all(mycursor, connection)

shop_dao = ShopDAO()
mycursor, connection = shop_dao.get_cursor()

# First database table 'productdata'
sql= '''
    CREATE TABLE IF NOT EXISTS productdata (
        ID INT AUTO_INCREMENT PRIMARY KEY,
        Product VARCHAR(300),
        Model VARCHAR(200),
        Price INT
    )
'''
mycursor.execute(sql)

# Second database table 'orderdata'
order_sql = '''
    CREATE TABLE IF NOT EXISTS orderdata (
        OrderID INT AUTO_INCREMENT PRIMARY KEY,
        ProductID INT,
        Quantity INT,
        TotalPrice INT,
        FOREIGN KEY (ProductID) REFERENCES productdata(ID)
    )
'''
# Execute the SQL statement
mycursor.execute(order_sql)
connection.commit()
shop_dao.close_all(mycursor, connection)

File: php/shop_DAO.php
<?php
class ShopDAO {
    private $db_host = 'localhost';
    private $db_name = 'wsaa1';
    private $db_user = 'root';
    private $db_pass = 'pass';

    private function getConnection() {
        $conn = new mysqli($this->db_host, $this->db_user, $this->db_pass, $this->db_name);
        if ($conn->connect_error) {
            die("Connection failed: " . $conn->connect_error);
        }
        return $conn;
    }

    public function get_cursor() {
        return $this->getConnection();
    }

    public function close_all($mycursor) {
        $mycursor->close();
    }

    // Create a new product
    public function create_product($product, $model, $price) {
        $conn = $this->getConnection();
        $stmt = $conn->prepare("INSERT INTO productdata (Product, Model, Price) VALUES (?, ?, ?)");
        $stmt->bind_param("ssd", $product, $model, $price);
        $stmt->execute();
        $product_id = $stmt->insert_id;
        $stmt->close();
        $conn->close();
        return ['id' => $product_id, 'Product' => $product, 'Model' => $model, 'Price' => $price];
    }

    // Get all products
    public function get_all_products() {
        $conn = $this->getConnection();
        $result = $conn->query("SELECT * FROM productdata");
        $products = [];
        while ($row = $result->fetch_assoc()) {
            $products[] = $row;
        }
        $conn->close();
        return $products;
    }

    // Get product by ID
    public function get_product_by_id($id) {
        $conn = $this->getConnection();
        $stmt = $conn->prepare("SELECT * FROM productdata WHERE id = ?");
        $stmt->bind_param("i", $id);
        $stmt->execute();
        $result = $stmt->get_result();
        $product = $result->fetch_assoc();
        $stmt->close();
        $conn->close();
        return $product;
    }

    // Update a product
    public function update_product($id, $product, $model, $price) {
        $conn = $this->getConnection();
        $stmt = $conn->prepare("UPDATE productdata SET Product = ?, Model = ?, Price = ? WHERE id = ?");
        $stmt->bind_param("ssdi", $product, $model, $price, $id);
        $stmt->execute();
        $stmt->close();
        $conn->close();
        return ['id' => $id, 'Product' => $product, 'Model' => $model, 'Price' => $price];
    }

    // Delete a product
    public function delete_product($id) {
        $conn = $this->getConnection();
        $stmt = $conn->prepare("DELETE FROM productdata WHERE id = ?");
        $stmt->bind_param("i", $id);
        $stmt->execute();
        $stmt->close();
        $conn->close();
        return true;
    }


    public function create_order($product_id, $quantity, $total_price) {
        $conn = $this->getConnection();
        $stmt = $conn->prepare("INSERT INTO orderdata (product_id, quantity, total_price) VALUES (?, ?, ?)");
        $stmt->bind_param("iid", $product_id, $quantity, $total_price);
        $stmt->execute();
        $order_id = $stmt->insert_id;
        $stmt->close();
        $conn->close();
        return ['order_id' => $order_id, 'product_id' => $product_id, 'quantity' => $quantity, 'total_price' => $total_price];
    }
}

$shop_dao = new ShopDAO();
?>
